### PR TITLE
[KO-321] [KO-320] Added a flag to disable PDB

### PR DESCRIPTION
**Pull request by sud82**
Added tests for disablePDB flag

### CODE REVIEW

File: test/poddisruptionbudget_test.go [hunk 1]
```diff
@@ -16,7 +18,8 @@ var _ = Describe(
 	"PodDisruptionBudget", func() {
 		ctx := context.TODO()
 		aeroCluster := &asdbv1.AerospikeCluster{}
-		maxAvailable := intstr.FromInt(0)
+		maxUnavailable := intstr.FromInt(0)
+		defaultMaxUpavailable := intstr.FromInt(1)
```
Comment (tanmayja) [inline test/poddisruptionbudget_test.go:22]: spelling

File: controllers/poddistruptionbudget.go
Comment (tanmayja) [inline controllers/poddistruptionbudget.go:1]: Certainly, adding a label to the PDB (Pod Disruption Budget) created by the operator can be a helpful approach to distinguish it from those created by users, especially if they share the same name as the cluster. This labeling practice can prevent accidental interference with user-created PDBs and ensure clarity in management and troubleshooting processes.

File: controllers/poddistruptionbudget.go [hunk 2]
```diff
@@ -14,6 +14,45 @@ import (
...
+		if errors.IsNotFound(err) {
+			// PodDisruptionBudget is already deleted
+			return nil
+		}
+
+		return err
+	}
+
+	// Delete the PodDisruptionBudget
+	return r.Client.Delete(context.TODO(), pdb)
```
Comment (abhishekdwivedi3060) [inline controllers/poddistruptionbudget.go:53]: We don't have to actually do get opertion. We can directly call delete and ignore isNotFound error.

File: test/poddisruptionbudget_test.go [hunk 3]
```diff
@@ -33,46 +36,97 @@ var _ = Describe(
...
+				Expect(err).To(HaveOccurred())
+				Expect(errors.IsNotFound(err)).To(BeTrue())
+				pkgLog.Info("PDB not created as expected")
+			})
+
+			It("Validate update disablePDB, the Operator will delete and recreate PDB", func() {
+				By("Create cluster with PDB enabled")
+				err := deployCluster(k8sClient, ctx, aeroCluster)
+				Expect(err).ToNot(HaveOccurred())
+				validatePDB(ctx, aeroCluster, 1)
```
Comment (abhishekdwivedi3060) [inline test/poddisruptionbudget_test.go:82]:
```suggestion
				validatePDB(ctx, aeroCluster, defaultMaxUpavailable.IntValue())
```

File: test/poddisruptionbudget_test.go [hunk 4]
```diff
@@ -33,46 +36,97 @@ var _ = Describe(
...
+
+				By("Update disablePDB to false")
+				aeroCluster, err = getCluster(k8sClient, ctx, clusterNamespacedName)
+				Expect(err).ToNot(HaveOccurred())
 
+				aeroCluster.Spec.DisablePDB = ptr.To(false)
 				err = updateCluster(k8sClient, ctx, aeroCluster)
 				Expect(err).ToNot(HaveOccurred())
-				validatePDB(ctx, aeroCluster, 0)
+				validatePDB(ctx, aeroCluster, 1)
```
Comment (abhishekdwivedi3060) [inline test/poddisruptionbudget_test.go:105]:
```suggestion
				validatePDB(ctx, aeroCluster, defaultMaxUpavailable.IntValue())
```

**Review by abhishekdwivedi3060 — CHANGES_REQUESTED** [review]

**Review by tanmayja — APPROVED** [review]

**Review by abhishekdwivedi3060 — APPROVED** [review]

**Pull request merged**